src/features/defi/components/Deposit/PairDeposit.tsx

```diff
@@ -0,0 +1,263 @@
+import { Button, Stack, useColorModeValue } from '@chakra-ui/react'
+import { Asset } from '@shapeshiftoss/asset-service'
+import { MarketData } from '@shapeshiftoss/types'
+import get from 'lodash/get'
+import { ControllerProps, useController, useForm, useWatch } from 'react-hook-form'
+import { useTranslate } from 'react-polyglot'
+import { Amount } from 'components/Amount/Amount'
+import { AssetIcon } from 'components/AssetIcon'
+import { AssetInput } from 'components/DeFi/components/AssetInput'
+import { FormField } from 'components/DeFi/components/FormField'
+import { Row } from 'components/Row/Row'
+import { Text } from 'components/Text'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+
+type DepositProps = {
+  asset1: Asset
+  asset2: Asset
+  destAsset: Asset
+  // Estimated apy (Deposit Only)
+  apy: string
+  // Users available amount
+  cryptoAmountAvailable1: string
+  cryptoAmountAvailable2: string
+  // Validation rules for the crypto input
+  cryptoInputValidation1?: ControllerProps['rules']
+  cryptoInputValidation2?: ControllerProps['rules']
+  // Users available amount
+  fiatAmountAvailable1: string
+  fiatAmountAvailable2: string
+  // Validation rules for the fiat input
+  fiatInputValidation1?: ControllerProps['rules']
+  fiatInputValidation2?: ControllerProps['rules']
+  // enables slippage UI (defaults to true)
+  enableSlippage?: boolean
+  // Asset1 market data
+  marketData1: MarketData
+  // Asset2 market data
+  marketData2: MarketData
+  // Array of the % options
+  percentOptions: number[]
+  isLoading: boolean
+  onContinue(values: DepositValues): void
+  onBack?(): void
+  onCancel(): void
+  syncPair?: boolean
+}
+
+enum Field {
+  FiatAmount1 = 'fiatAmount1',
+  FiatAmount2 = 'fiatAmount2',
+  CryptoAmount1 = 'cryptoAmount1',
+  CryptoAmount2 = 'cryptoAmount2',
+}
+
+export type DepositValues = {
+  [Field.FiatAmount1]: string
+  [Field.CryptoAmount1]: string
+  [Field.FiatAmount2]: string
+  [Field.CryptoAmount2]: string
+}
+
+function calculateYearlyYield(apy: string, amount: string = '') {
```

Comment:
We already have this in `src/plugins/cosmos/components/modals/Staking/StakingCommon.ts`